snapcraft/internal/errors.py

```diff
@@ -216,6 +216,34 @@ def __init__(self, filename, message):
         super().__init__(filename=filename, message=message)
 
 
+class InvalidDesktopEntry(SnapcraftError):
+
+    fmt = (
+        'Failed to find any Desktop Entry: '
+        'No Desktop Entry for {filename!r}: {message}.'
+        # FIXME include how to fix each of the possible desktop file errors.
+        # https://bugs.launchpad.net/snapcraft/+bug/1727435
+        # --elopio - 2017-10-25
+    )
+
+    def __init__(self, filename, message):
+        super().__init__(filename=filename, message=message)
+
+
+class MissingExecKey(SnapcraftError):
+
+    fmt = (
+        'Failed to find Exec line: '
+        'No Exec line for {filename!r}: {message}.'
+        # FIXME include how to fix each of the possible desktop file errors.
+        # https://bugs.launchpad.net/snapcraft/+bug/1727435
+        # --elopio - 2017-10-25
```

Comment:
Again, this exception is suitably specific-- just put the message here in the exception. Once you do that, this comment is no longer valid.